component-tools/src/main/java/org/talend/sdk/component/tools/validator/ActionValidator.java

```diff
@@ -206,6 +213,44 @@ private Stream<String> findDiscoverSchemaExtendedErrors(final AnnotationFinder f
                 .orElseGet(Stream::empty);
     }
 
+    /**
+     * Checks method signature for @DynamicDependencies annotation.
+     * Valid signatures are:
+     * <ul>
+     * <li>public List<String> getDependencies(@Option("configuration") final TheDataset dataset)</li>
+     * </ul>
+     *
+     * @param finder
+     * @return Errors on @DynamicDependencies method
+     */
+    private Stream<String> findDynamicDependenciesErrors(final AnnotationFinder finder) {
+
+        final Stream<String> optionParameter = finder
+                .findAnnotatedMethods(DynamicDependencies.class)
+                .stream()
+                .filter(m -> !hasOption(m))
+                .map(m -> m + " should have a parameter being an option (marked with @Option)")
+                .sorted();
+
+        final Stream<String> returnType = finder
+                .findAnnotatedMethods(DynamicDependencies.class)
+                .stream()
+                .filter(m -> !hasStringInList(m))
+                .map(m -> m + " should return List<String>")
+                .sorted();
+
+        final Stream<String> dataset = finder
+                .findAnnotatedMethods(DynamicDependencies.class)
+                .stream()
+                .filter(m -> !hasTypeParameter(m, DataSet.class))
+                .map(m -> m + " should have its Dataset parameter")
+                .sorted();
```

Comment:
Sorry, did not noticed at first review: Also this one can be coupled with `optionParameter`. We need an `@Option`ed dataset...